src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -130,6 +142,9 @@ export async function getDataSourceById(
   savedObjectsClient: SavedObjectsClientContract
 ) {
   return savedObjectsClient.get('data-source', id).then((response) => {
+    if (!response || response.error) {
+      throw new Error('Unable to find data source');
```

Comment:
does the get data source API throw exception when failed to get data source or do we need to handle it this way?

Reply:
If that's the case, then we don't need the try catch block, right?